### ISSUE TITLE
[Proposal] Vietnam Geeky Community Wiki

### ISSUE THREAD

**Issue by huytd** [geeky-vietnam/geeky.vn#1]
# Proposal: Vietnam Geeky Community Wiki

The idea is to use [Github wiki](https://github.com/geeky-vietnam/geeky.vn/wiki) to run a community-driven wiki, where anyone can read/write/create a new technical article for the community, with the help from the community to review/edit/validate the knowledge in that article.

## Motivation and Problems

**1) The current situation of Vietnam tech community:** We are having so many tech blogs (Kipalog, Viblo, Machine Learning Cho Nguoi Moi Bat Dau,...) and thousands of personal technical blogs (Toi Di Code Dao, Codeaholicguy, The Full Snack :trollface:,...).

All of them has the same problem: **Each blog post was created by a single person and the knowledge in each post are limited to the knowledge of the author.**

There is nothing to guarantee that the knowledge is always valid. And it's even worse if the author posts some wrongful knowledge on the internet. Without carefully tested, validated. Many other people who read that post will get the wrong idea as well.

**2) What about the comment feature?** Vietnamese technical communities are not **really** good at healthy debating or giving feedbacks. There are thousands of reader who only read and like, then share without thinking about the validity of what they read.

Some people may recognize the problem in a blog post and leave the comment but updating the post or not, is depends on the author.

Once again, Vietnamese authors are not really open for public comments.

**3) Most of the young geeks are bad at English**. This is true, and we cannot simply ask them to learn English before they start jumping to this industry. It takes time for them, we have to admit it. The lack of English skill will prevent them from reading foreign technical materials such as blogs, books,...

## Solution

People are avoiding public discussing but they are very open for silence-feedback (or private messages, contributing,...), so it's a good idea to have a technical wiki.

With this wiki, we can easily solve the **problem `#1`**, everyone can contribute a new technical article, the community will help review/improve it. And everything will be controlled by Github Wiki (edit history, pull request,...). Make sure all of the contents are the best.

Speaking of **problem `#2`**, it doesn't matter if nobody gives a comment, they can just edit it right away if they feel there are something wrong in the post.

For **problem `#3`**, because all the posts are in Vietnamese, young geeks are free to read them, the community will help translate great-good materials from other languages, or one can improve their English skill by reading/translating and contribute those materials to the wiki.

## How we gonna do it?

We use Github Wiki. No need for starting any new coding project.

## Known issues

1) Need to have a team to run the wiki (merging/reviewing edits/pull requests,...)
2) _Nothing else_

**Comment by hieuk09** [comment]
> Speaking of problem #2, it doesn't matter if nobody gives a comment, they can just edit it right away if they feel there are something wrong in the post.

We'll do it in wiki ways, maintainers review all the changes and make sure it correct?

**Comment by huytd** [comment]
Yes, that's it. At least, in this way, we will have a reviewer for every content on the wiki.

**Comment by runlevel5** [comment]
@huytd What wiki software do you think that is suitable? If GitHub Wiki is one, why is it?

**Comment by huytd** [comment]
I choose Github Wiki because it's very easy to use. Friendly UI and familiar with most of the developers.

Moreover, if somebody submits a change on Github Wiki, they have to create a pull request. With this workflow, it can help the maintainers a lot.

The other wiki software is not built for developers in the first place, so we need more learning curves and adoption as well, which will take time.

**Comment by runlevel5** [comment]
@huytd can you search with GH Wiki?

**Comment by huytd** [comment]
[Yes you can](https://github.com/blog/2219-search-wiki-pages).

**Comment by runlevel5** [comment]
@huytd let's wait after we launch the website. I want to observe Grokking Forum and learn from how we could curate / collaborate to prepare the best content, would wiki be the best solution out there?
